### PR TITLE
Add qubit loss support in QuantumErrorProfile and noise config support in qdk sparse state simulator

### LINKED ISSUE #416: Deduce `num_qubits` in `GateErrorDef` from standard gate set
The current implementation of `GateErrorDef` takes [`num_qubits`](https://github.com/microsoft/qdk-chemistry/blob/acf17055a53b535c2e9a20ea822ea20256a55a25/python/src/qdk_chemistry/data/noise_models.py#L124) as an explicit argument in the dict specification. This puts an undue burden on the user to manually specify noise profiles. This should instead be deduced from the standard set of gates to improve UX

### PR DESCRIPTION
**Pull request by RushiGong**

### CODE REVIEW

File: python/src/qdk_chemistry/algorithms/circuit_executor/qdk.py [hunk 1]
```diff
@@ -87,7 +87,8 @@ def _run_impl(
         )
         Logger.debug(f"Measurement results obtained: {raw_results}")
         # Reorder bits in each measurement result to match Little Endian convention
-        bitstrings = ["".join("0" if str(x) == "Zero" else "1" for x in reversed(one_run)) for one_run in raw_results]
+        # If result is Loss, treat it as "0" in the bitstring
+        bitstrings = ["".join("1" if str(x) == "One" else "0" for x in reversed(one_run)) for one_run in raw_results]
         counts = dict(Counter(bitstrings))
```
[the remark this answers is not in the document]
Comment (RushiGong) [reply python/src/qdk_chemistry/algorithms/circuit_executor/qdk.py]: Separate out the `Loss`

File: python/src/qdk_chemistry/plugins/qiskit/_interop/noise_model.py [hunk 2]
```diff
@@ -31,19 +31,32 @@ def get_noise_model_from_profile(
...
     return noise_model
+
+
+def _check_num_qubits(gate: str, noise_model: NoiseModel) -> int:
+    """Helper function to determine the number of qubits for a given gate."""
+    if gate in noise_model._1qubit_instructions:  # noqa: SLF001
+        return 1
+    if gate in noise_model._2qubit_instructions:  # noqa: SLF001
+        return 2
+    if gate in noise_model._3qubit_instructions:  # noqa: SLF001
```
[the remark this answers is not in the document]
Comment (RushiGong) [reply python/src/qdk_chemistry/plugins/qiskit/_interop/noise_model.py:55]: Use qiskit's definition for their noise model.

File: python/src/qdk_chemistry/plugins/qiskit/_interop/noise_model.py [hunk 3]
```diff
@@ -31,19 +31,32 @@ def get_noise_model_from_profile(
...
+        return 3
+    raise ValueError(f"Unsupported gate type: {gate}")
```
[the remark this answers is not in the document]
Comment (RushiGong) [reply python/src/qdk_chemistry/plugins/qiskit/_interop/noise_model.py:57]: Required to error out.

File: python/src/qdk_chemistry/data/noise_models.py [hunk 4]
```diff
@@ -152,55 +141,43 @@ def __init__(
...
+                    self.one_qubit_gates.add(gate)
+                elif gate in TWO_QUBIT_SUPPORTED_GATES:
+                    self.two_qubit_gates.add(gate)
+                elif gate in THREE_QUBIT_SUPPORTED_GATES:
+                    self.three_qubit_gates.add(gate)
+                validated_rates: dict[SupportedErrorTypes, float] = {}
+                for error_type_key, rate in error_rates.items():
+                    error_type = (
+                        error_type_key
+                        if isinstance(error_type_key, SupportedErrorTypes)
```
[the remark this answers is not in the document]
Comment (RushiGong) [reply python/src/qdk_chemistry/data/noise_models.py:165]: Already a dictionary.

File: python/src/qdk_chemistry/data/noise_models.py [hunk 5]
```diff
@@ -414,23 +381,26 @@ def to_qdk_noise_config(self) -> NoiseConfig:
...
-                    getattr(noise, gate_name_qdk).set_depolarizing(error_def["rate"])
+                    if error_type == SupportedErrorTypes.DEPOLARIZING_ERROR:
+                        getattr(noise, gate_name_qdk).set_depolarizing(rate)
+                    elif error_type == SupportedErrorTypes.QUBIT_LOSS:
+                        getattr(noise, gate_name_qdk).loss = rate
+                    else:
+                        raise ValueError(f"Error type {error_type} is not currently supported.")
                 except AttributeError:
                     # Warn and skip unsupported gates
                     Logger.warn(f"Gate {gate_name} not supported in QDK noise config; skipping.")
```
[the remark this answers is not in the document]
Comment (RushiGong) [reply python/src/qdk_chemistry/data/noise_models.py]: Fixed.

File: python/src/qdk_chemistry/algorithms/circuit_executor/qdk.py [hunk 6]
```diff
@@ -138,36 +144,49 @@ def _run_impl(
...
             bitstrings = [
-                "".join("0" if str(x) == "Zero" else "1" for x in reversed(one_run)) for one_run in raw_results
+                "".join("1" if str(x) == "One" else ("L" if str(x) == "Loss" else "0") for x in reversed(one_run))
+                for one_run in raw_results
             ]
-            bitstring_counts = dict(Counter(bitstrings))
+            clean = [b for b in bitstrings if "L" not in b]
+            lost = [b for b in bitstrings if "L" in b]
+            bitstring_counts = dict(Counter(clean))
+            loss_bitstrings = dict(Counter(lost)) if lost else None
```
Comment (wavefunction91) [inline python/src/qdk_chemistry/algorithms/circuit_executor/qdk.py]: This code is repeated a few places, can we encapsulate it somewhere?

File: python/src/qdk_chemistry/data/circuit_executor_data.py [hunk 7]
```diff
@@ -31,21 +31,25 @@ def __init__(
...
-            bitstring_counts (dict[str, int] | None): Bitstring count dict.
-            total_shots (int | None): Total number of shots used for the measurement.
-            executor (str | None): Name of the executor used for the measurement.
-            executor_metadata (Any | None): Metadata associated with the executor.
+            bitstring_counts: Bitstring count dict (clean shots only, 0/1).
+            total_shots: Total number of shots used for the measurement.
+            executor: Name of the executor used for the measurement.
+            executor_metadata: Metadata associated with the executor.
+            loss_bitstrings: Bitstring counts for shots where at
+                least one qubit was lost. Uses 'L' to mark lost qubits (e.g. '0L10').
```
Comment (wavefunction91) [inline python/src/qdk_chemistry/data/circuit_executor_data.py]: I feel like this needs more context - qubit loss is very niche as an error channel

File: python/src/qdk_chemistry/data/noise_models.py [hunk 8]
```diff
@@ -131,11 +118,12 @@ class QuantumErrorProfile(DataClass):
...
-        description (str): Description of what the error profile represents.
-        errors (dict[SupportedGate, GateErrorDef]): Dictionary mapping gate names to their error properties.
-        one_qubit_gates (list[str]): List of gate names that operate on a single qubit.
-        two_qubit_gates (list[str]): List of gate names that operate on two qubits.
+        name: Name of the quantum error profile.
+        description: Description of what the error profile represents.
+        errors: Dictionary mapping gate names to their error properties.
+        one_qubit_gates: Set of gate names that operate on a single qubit.
+        two_qubit_gates: Set of gate names that operate on two qubits.
+        three_qubit_gates: Set of gate names that operate on three qubits.
```
Comment (wavefunction91) [inline python/src/qdk_chemistry/data/noise_models.py]: Make these a map or a getter. Enumerating these kinds of things explicitly just complicates the API

**Review by wavefunction91 — CHANGES_REQUESTED** [review]

File: python/src/qdk_chemistry/data/circuit_executor_data.py (hunk 7 above)
Comment (RushiGong) [reply python/src/qdk_chemistry/data/circuit_executor_data.py]: Docstring has been improved.

File: python/src/qdk_chemistry/algorithms/circuit_executor/qdk.py (hunk 6 above)
Comment (RushiGong) [reply python/src/qdk_chemistry/algorithms/circuit_executor/qdk.py]: A helper function is adeed.

**Review by wavefunction91 — APPROVED** [review]
